test/environments/norails/Gemfile

```diff
@@ -6,7 +6,7 @@ gem 'rake', '~> 12.3.3'
 gem 'minitest', '5.3.3'
 gem 'minitest-stub-const', '~> 0.6'
 gem 'mocha', '~> 1.9.0', :require => false
-gem 'rack', '>= 2.1.4'
+gem 'rack', "~> #{RUBY_VERSION >= '2.3.0' ? '2.2.4' : '2.1.4.1'}"
```

Comment:
Does this prevent us from testing Rack 3 in `norails`?

Reply:
This change was intended to just be more specific with the Ruby 2.2 and Ruby 2.3+ specific versions of Rack 2. It's the previous existence of the `rack-test` constraint to a super old version (`< 0.8.0`) which was already keeping us on Rack 2 / `< Rack 3`.